src/Workspaces/Core/Portable/ExternalAccess/Pythia/Api/PythiaObjectPool.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Pythia.Api
+{
+    internal static class PythiaObjectPool
```

Comment:
⚠️  External code should not be using our object pools. Is this absolutely required, or is it sufficient to copy the sources for the pool into the external library for full encapsulation?

Reply:
So, ideally Pythia would use Microsoft.CodeAnalysis.PooledObject source package. The problem is that since it has IVT to Roslyn the compiler sees both source implementation and implementation from metadata and reports warning. We can address that in two ways 1) make a copy and rename the namespace, 2) use extern alias for Roslyn references.

Reply:
I'll keep it this way for now as it preserves the current behavior. We can remove later.